test/single/test_elastic_discovery.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2022 Uber Technologies, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import mock
+from parameterized import parameterized
+
+from horovod.runner.elastic.discovery import HostDiscoveryScript
+
+# Mocked default slots for testing.
+DEFAULT_SLOTS = mock.Mock()
+
+
+class HostDiscoveryScriptTests(unittest.TestCase):
+    """
+    Tests for host discovery script in horovod.elastic.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(HostDiscoveryScriptTests, self).__init__(*args, **kwargs)
+
+    @parameterized.expand([
+        ["", {}],  # No host found.
+        ["host-1", {"host-1": DEFAULT_SLOTS}],
+        ["host-1:2", {"host-1": 2}],
+        ["host-1\nhost-5", {"host-1": DEFAULT_SLOTS, "host-5": DEFAULT_SLOTS}],
+        ["host-1:2\nhost-4:3", {"host-1": 2, "host-4": 3}],
```

Comment:
Can we also add tests where multiple hosts exist, but also an empty line, because the new code should handle that case gracefully as well:

    ["\nhost-1:2\nhost-4:3", ...],
    ["host-1:2\n\nhost-4:3", ...],
    ["host-1:2\nhost-4:3\n", ...],

Reply:
Sure! I added those cases. Please review again. Thanks!